app/forms/participant_mentor_form.rb

```diff
@@ -12,7 +12,9 @@ def mentor
   end
 
   def available_mentors
-    school.mentors.order(:full_name)
+    User.where(
+      id: school.mentor_profiles_for(Cohort.find(cohort_id)).select(:user_id),
+    ).order(:full_name)
```

Comment:
Soo, this is not really something for this PR, but in general, I have a feeling most of the time when school users do an action, they really operate on a school_cohort. That's why we keep passing both schools and cohorts around. 

I wonder if we should shift our approach to suit that - have a bunch of things like profiles point at school cohorts. And if they need school or cohort, they can fetch it `through`.

If we do that, this could be a lot simpler. Something like `school_cohort.mentors.order(:full_name)`. Thoughts?

Reply:
Yep, see https://dfedigital.atlassian.net/browse/CPDRP-545